web/src/emoji_picker.js

```diff
@@ -673,20 +681,7 @@ export function toggle_emoji_popover(element, id, coming_from_actions_popover) {
         return;
     }
 
-    const $elt = $(element);
-    $elt.closest(".message_row").toggleClass("has_popover has_emoji_popover");
-    if (id !== undefined) {
-        message_lists.current.select_id(id);
-    }
-
-    if (user_status_ui.user_status_picker_open()) {
-        build_emoji_popover($elt, id);
```

Comment:
It seems like this commit deletes the special logic for the `user_status_ui.user_status_picker_open` case without explanation for why that might not be correct. I suspect this is wrong -- I don't think `reaction_button_visible` is a class that's intended to be toggled on that element?

In general this commit would have been much easier to review if you broke it up into pieces -- the parts that are just moving code and easily verified as equivalent with a few words of text explaining why that's the case first, and then anything like that this that might be a functional change with a clear explanation in its commit message for why it's correct.

Reply:
Testing, I find that the user status picker seems to mostly work but if you click the button to pop it 3 times in a row, it doesn't toggle properly -- the second click doesn't do anything visible, so it's definitely bugged.

Reply:
> Testing, I find that the user status picker seems to mostly work but if you click the button to pop it 3-4 times in a row, it doesn't toggle properly -- the first click opens, second closes, and third just appears the tooltip for the element, so it's definitely bugged somehow.

Fixed this in [2fb09b11eca93f3c04d37329139ac0be0176aaf2](https://github.com/zulip/zulip/pull/26354/commits/2fb09b11eca93f3c04d37329139ac0be0176aaf2)

Reply:
> In general this commit would have been much easier to review if you broke it up into pieces -- the parts that are just moving code and easily verified as equivalent with a few words of text explaining why that's the case first, and then anything like that this that might be a functional change with a clear explanation in its commit message for why it's correct.

Got it!

Reply:
> It seems like this commit deletes the special logic for the user_status_ui.user_status_picker_open case without explanation for why that might not be correct.

Yeah, it should be a separate commit to show the reviewer that it's a functional change.

Now we add the "reaction_button_visible" class to every reference element of the emoji popover. There's no need to handle the user status picker case since the reference element should always be visible. That's why we're reducing some complexity, even though it means adding a class to the user status picker reference.

Reply:
OK. Please add the explanation you wrote here to the relevant commit message.

Reply:
Added!